radix_sort.py

```diff
@@ -0,0 +1,86 @@
+import timeit
+import random
+
+
+def r_sort(alist, base=10):
+
+    def list_to_buckets(alist, base, iteration):
+        buckets = [[] for i in range(base)]
+        for number in alist:
+            digit = (number // (base ** iteration)) % base
+            buckets[digit].append(number)
+        return buckets
+
+    def buckets_to_list(buckets):
+        alist = []
+        for bucket in buckets:
+            alist.extend(bucket)
+        return alist
+
+    max_val = 0
+    for i in alist:
+        if i > max_val:
+            max_val = i
+
+    it = 0
+
+    while ((base ** it) <= max_val):
```

Comment:
interesting!  So you are using the exponent to determine the number of iterations to make.  I have not seen this approach yet.  I like it.